remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+bedtools getfasta -fi $oldgenome -bed flanking.corrected.bed -fo variants_reads.fa
+
+#replace the colon separators with "|"
+sed -i 's/:/|/' variants_reads.fa
+
+#store rsIDs
+awk '{print $4}' flanking.corrected.bed > rsIDs.txt
+
+#store variant bases
+awk '{print $7}' flanking.corrected.bed > variant_bases.txt
+
+#store the other vcf columns
+awk '{print $5, $8, $9}' flanking.corrected.bed > qual_filt_info.txt
+
+#paste the names, variant bases, then fasta sequences into a new file
+paste <(grep '^>' variants_reads.fa) variant_bases.txt rsIDs.txt qual_filt_info.txt <(grep -v '^>' variants_reads.fa) > temp.txt
+rm flanking.corrected.bed variant_bases.txt rsIDs.txt qual_filt_info.txt  variants_reads.fa
+
+#reformat the fasta ID
+sed 's/\t/|/g; s/ /|/g; s/\(.*\)|/\1 /' temp.txt | tr ' ' '\n' > variant_reads.out.fa
+rm temp.txt
+
+
+
+
+
+
+#2) Mapping with bowtie2
+echo "---------------------------2) Mapping with bowtie2---------------------------"
+#"index" is the prefix of the output filenames, not a directory
+echo "#### Building index: may take a while ####"
+bowtie2-build $newgenome index
+echo "#### Mapping results: ####"
+bowtie2 -f -x index variant_reads.out.fa | samtools view -bS - > reads_aligned.bam
+#example output:
+#>1094 reads; of these:
+#>  1094 (100.00%) were unpaired; of these:
+#>   7 (0.64%) aligned 0 times
+#>    48 (4.39%) aligned exactly 1 time
+#>    1039 (94.97%) aligned >1 times
+#>99.36% overall alignment rate
+
+
+
+
+
+#3) Data extraction
+echo '------------------------------3) Data extraction-----------------------------'
+#extract chromosome, rsID, position on new genome, original variant allele, qual, filter and info fields
+samtools sort reads_aligned.bam -o reads_aligned.sorted.bam
+samtools index reads_aligned.sorted.bam
+./reverse_strand.py -i reads_aligned.sorted.bam -o variants_remapped.vcf
+rm reads_aligned.bam reads_aligned.sorted.bam reads_aligned.sorted.bam.bai
+
+#add interval for variant position in bed format (required by getfasta)
+awk '{print $1"\t"$2-1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > variants_remapped.bed
+
+#getfasta to get the REF genome alleles and then extract the genome alleles without the positions
+bedtools getfasta -tab -fi $newgenome -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+rm variants_remapped.bed
+#make all alleles caps (sometimes there are lower-case bases)
+tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+
+#paste all the information into the correct columns
+awk '{print $1"\t"$2"\t"$3}' variants_remapped.vcf > temp_first_3_columns.txt
+awk '{print $4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > temp_last_4_columns.txt
+paste temp_first_3_columns.txt genome_alleles.fixed.txt temp_last_4_columns.txt > var_pre_final.vcf
+rm temp_first_3_columns.txt temp_last_4_columns.txt genome_alleles.txt variants_remapped.vcf genome_alleles.fixed.txt
+
+
+#header:
+#create list of contigs/chromosomes to be added to the header
```

Comment:
I suggest you look into `bcftools reheader` to update the header information. Among other things, it can automatically update the list of contigs based on fai index. This would also the problem in current implementation where VCF header will only include contigs which have at least one variant on them (in reality it should include **all** contigs present in the assembly)

Reply:
This has been discussed and we established that we can't use `bcftools reheader` in its current version as it does not have the functionality we need, so I'll continue using the bash commands to recreate the header.